### ISSUE TITLE
After recent plugin update Subclipse crashes Eclipse IDE when accessing configuration properties

### ISSUE THREAD

**Issue by JRSofty** [subclipse/subclipse#131]
Started my Eclipse this morning and was notified that there was an update for either SVNKit or Subclipse itself. After the IDE restarted and I attempted to access the SVN properties to check configuration the IDE just disappeared, and when I looked in the folder I found a Java Crash log (see attached) Also when I attempted to perform an update from repository the IDE crashed.

I was able to correct the problem by looking into the plugins folder and identifying that two files were added today for the JavaHL. Once I removed these files and restarting the IDE I was able to use the SVN features again.

[hs_err_pid7036.log](https://github.com/subclipse/subclipse/files/4572932/hs_err_pid7036.log)
[hs_err_pid10312.log](https://github.com/subclipse/subclipse/files/4572933/hs_err_pid10312.log)
[hs_err_pid10972.log](https://github.com/subclipse/subclipse/files/4572934/hs_err_pid10972.log)

**Comment by maurorulli** [comment]
@JRSofty pelase can you specify which files to remove?

**Comment by JRSofty** [comment]
Hello @maurorulli, 

Sure there were two 
* org.tigris.subversion.clientadapter.javahl.win64_<version 1.14 I believe>.jar
* org.tigris.subversion.clientadapter.javahl_<version 1.14 I believe>.jar

In both cases I had two copies of each file with different version numbers. The 1.11.1 version was updated last in November 2019, and the others were updated today (when I performed my update). Removing the two newer files and keeping my older files allowed the Subclipse to work without crashing my IDE. All of these files are under the eclipse/plugins folder.

**Comment by maurorulli** [comment]
In my eclipse/plugins folder there is just one file:
org.eclipse.equinox.launcher_1.5.700.v20200207-2156.jar

**Comment by JRSofty** [comment]
@maurorulli Look to where your eclipse is installed, basically where you will find eclipse.exe or if you are working in linux the executable eclipse file. In this directory there are different folders like:

* configuration
* dropins
* features
* p2
* plugins
* readme

In this plugins folder you should look.

**Comment by PyvesB** [comment]
Saw the update pop up this morning as well and came looking here for a changelog. The problem is likely due to recent changes in the [JavaHL library](https://github.com/subclipse/javahl) to support SVN 1.14. Will postpone update for now.

**Comment by maurorulli** [comment]
> @maurorulli Look to where your eclipse is installed, basically where you will find eclipse.exe or if you are working in linux the executable eclipse file. In this directory there are different folders like:
> 
> * configuration
> * dropins
> * features
> * p2
> * plugins
> * readme
> 
> In this plugins folder you should look.

This is exactly where I'm looking for, I have the following 4 folders:

- configuration
- dropins
- plugins
- readme
then eclipse.ini and eclipse.exe

**Comment by JRSofty** [comment]
@maurorulli 

Strange. I'm not sure why you don't see the files in your plugins folder. You might need someone else's help on where to find where your plugins are installed.

**Comment by maurorulli** [comment]
> @maurorulli Look to where your eclipse is installed, basically where you will find eclipse.exe or if you are working in linux the executable eclipse file. In this directory there are different folders like:
> 
> * configuration
> * dropins
> * features
> * p2
> * plugins
> * readme
> 
> In this plugins folder you should look.

Ok, I've found .p2/pool/plugins folder under windows user folder. Here there are all plugins jars.
I've the following:
- org.tigris.subversion.clientadapter.javahl.win64_1.14.0.202005031741.jar
- org.tigris.subversion.clientadapter.javahl_1.14.0.jar
- org.tigris.subversion.clientadapter.javahl.win64_1.11.1.201901162041.jar
- org.tigris.subversion.clientadapter.javahl_1.11.1.jar
- org.tigris.subversion.clientadapter.svnkit_1.8.12.3.jar
- org.tigris.subversion.clientadapter_1.10.0.jar

Any ideas wich one I've to remove to solve?

**Comment by JRSofty** [comment]
These two

* org.tigris.subversion.clientadapter.javahl.win64_1.14.0.202005031741.jar
* org.tigris.subversion.clientadapter.javahl_1.14.0.jar

Then restart your IDE

**Comment by maurorulli** [comment]
> These two
> 
> * org.tigris.subversion.clientadapter.javahl.win64_1.14.0.202005031741.jar
> * org.tigris.subversion.clientadapter.javahl_1.14.0.jar
> 
> Then restart your IDE

Bingo.
Really thanks @JRSofty , I can finally start working today.
God make you immune to the virus.

**Comment by markphip** [comment]
Thanks for reporting.

I have rolled back the update .. this will not help anyone with the problem but it will at least stop the spread to others.  I tested this on Windows 10 before releasing it.  I will look at the files you provided and do some more testing later today.  The likely problem is a dependent DLL is missing or it loaded an incompatible version.

**Comment by markphip** [comment]
I think I know what the problem is .. and I was able to make my Eclipse crash. It manifests a bit differently for me though. My Eclipse would load but I had to do something like look at the SVN preferences to make it crash. I suspect you have something like the Synch view loaded which I do not and it is trying to run a SVN operation during startup.

Anyway .. when I can reproduce the problem it is related to the OpenSSL DLL's. It was working for me because I had the right version of them in PATH .. it is supposed to load them from the JavaHL plugin. The issue is that with OpenSSL 1.1.1 they use different name for the DLL on 32 vs 64-bit Windows and I did not notice that before.  So I have to modify the Java code to load the -x64 version of the DLL when on a 64-bit JVM. Working on that now ...

Workaround: Copy the 2 DLL's named "`libcrypto-1_1-x64.dll`" and "`libssl-1_1-x64.dll`" to a folder in PATH. You would have to get them out of a directory that is named something like this:  `C:\eclipse\configuration\org.eclipse.osgi\491\0\.cp\`
